docs/source/getting-started.md

```diff
@@ -157,10 +157,103 @@ If we press the play button in the middle, we should see a response on the right
 
 ![The response from our server shows title and author!](./images/getting-started/graphql-playground-response.png)
 
+## Server integrations
+
+Depending on whether we are creating a new application or an existing application, the steps will vary slightly since Apollo Server must adapt to the semantics of existing servers (e.g. Express, Hapi, etc.)
+
+### Middleware
+
+Existing applications generally already have middleware in place and Apollo Server works along with those middleware. To integrate with Apollo Server, we'll pass it into the `server.applyMiddleware` method as `app` to add the Apollo Server's middleware.
+
+> The existing application is frequently already named `app`, especially when using Express.  If the application is identified by a different variable, pass the existing variable in place of `app`.
+
+The following code uses the `apollo-server-express` package, which can be installed with `npm install apollo-server-express`.
+
+```js
+const { ApolloServer, gql } = require('apollo-server-express');
+const { typeDefs, resolvers } = require('./schema');
+
+const server = new ApolloServer({
+  // These will be defined for both new or existing servers
+  typeDefs,
+  resolvers,
+});
+
+server.applyMiddleware({ app }); // app is from an existing express app
+
+app.listen({ port: 4000 }, () =>
+  console.log(`🚀 Server ready at http://localhost:4000${server.graphqlPath}`)
+)
+```
+
+Hapi follows the same pattern with `apollo-server-express` replaced with `apollo-server-hapi` and `app` replaced with Hapi server. `applyMiddleware` registers plugins, so it should be called with `await`.
```

Comment:
This wasn't introduced by this PR (I don't think?) but, contextually, I find this note about Hapi a bit random unless it's trying to lay out an example of similar patterns being available for other frameworks.  If that's what it's doing though, I think it falls way short since the patterns are often quite a bit different from framework to framework.  I _think_ the actual `README.md` files within the [individual framework's directories within `packages/`](https://github.com/apollographql/apollo-server/tree/master/packages/) actually do lay out the more accurate steps, but I'm not positive if they're entirely up to date.

Regardless, this whole pattern will go away with Apollo Server 3.x if #3184 goes through (which I suspect it will, one way or another).

Reply:
Simplified this hapi language in the new article `middleware.md` that pulls that information out of the getting-started